api/clients/eigenda_client.go

```diff
@@ -200,10 +237,12 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 					alreadyWaitingForDispersal = true
 				}
 			case grpcdisperser.BlobStatus_FAILED:
-				errChan <- fmt.Errorf("EigenDA blob dispersal failed in processing, requestID=%s: %w", base64RequestID, err)
+				// TODO: when exactly does this happen? I think only happens if ethereum reorged and the blob was lost
```

Comment:
1. blob has expired, a client retrieve after 14 days. Sounds like 400 errors. (fine to ignore, since it would timeout already) 
2. internal logic error while requesting encoding (shouldn't happen), but it should return 503
3. wait for blob finalization from confirmation and blob retry has exceeded its limit. Thinking a bit more, it is possible that a chain re-org triggered a Failed status. See code (https://github.com/Layr-Labs/eigenda/blob/master/disperser/batcher/finalizer.go#L179-L189). So we should be returning 500, not 503. 

In any cases, EigenDA returned error into the failed status. So it makes sense that we return 500, i.e. internalServerError.

Reply:
Updated comments with your info: [585d68e](https://github.com/Layr-Labs/eigenda/pull/828/commits/585d68eb588fceebe90644810733d3373d83f371)